icu4c/source/test/intltest/restest.cpp

```diff
@@ -655,3 +664,160 @@ ResourceBundleTest::TestGetLocaleByType(void)
+    // This example is from the en_CA resource in the "unit" tree: The four test cases below show what we get when we call
+    // getStringWithFallback():
+    // - units/duration/day-person/other doesn't exist in either en_CA or en, so we fall back on root.
+    // - root/units aliases over to LOCALE/unitsShort.
+    // - unitsShort/duration/day-person/other also doesn't exist in either en_CA or en, so we fall back to root again.
+    // - root/unitsShort/duration/day-person aliases over to LOCALE/unitsShort/duration/day.
+    // - unitsShort/duration/day/other also doesn't exist in en_CA, so we fallback to en.
+    // - en/unitsShort/duration/day/other DOES exist and has "{0} days", so we return that.
+    // It's the same basic story for week-person, month-person, and year-person, except that:
+    // - unitsShort/duration/day doesn't exist at all in en_CA
+    // - unitsShort/duration/week DOES exist in en_CA, but only contains "per", so we inherit "other" from en
+    // - unitsShort/duration/month/other DOES exist in en_CA (it overrides "{0} mths" with "{0} mos")
+    // - unitsShort/duration/year DOES exist in en_CA, but only contains "per", so we inherit "other" from en
+    UErrorCode err = U_ZERO_ERROR;
+    LocalUResourceBundlePointer en_ca(ures_open(U_ICUDATA_UNIT, "en_CA", &err));
+    
+    if (!assertSuccess("Failed to load en_CA resource in units tree", err)) {
+        return;
+    }
+    assertEquals("Wrong result for units/duration/day-person/other", u"{0} days", ures_getStringByKeyWithFallback(en_ca.getAlias(), "units/duration/day-person/other", nullptr, &err));
+    assertEquals("Wrong result for units/duration/week-person/other", u"{0} wks", ures_getStringByKeyWithFallback(en_ca.getAlias(), "units/duration/week-person/other", nullptr, &err));
+    assertEquals("Wrong result for units/duration/month-person/other", u"{0} mos", ures_getStringByKeyWithFallback(en_ca.getAlias(), "units/duration/month-person/other", nullptr, &err));
+    assertEquals("Wrong result for units/duration/year-person/other", u"{0} yrs", ures_getStringByKeyWithFallback(en_ca.getAlias(), "units/duration/year-person/other", nullptr, &err));
+
+    // getAllChildrenWithFallback() wasn't bringing all of those things back, however.  When en_CA/units/year-person
+    // aliased over to en_CA/unitsShort/year, the sink would only be called on en_CA/unitsShort/year, which means it'd
+    // only see en_CA/unitsShort/year/per, because "per" was the only thing contained in en_CA/unitsShort/year.  But
+    // en_CA/unitsShort/year should be inheriting "dnam", "one", and "other" from en/unitsShort/year.
+    // getAllChildrenWithFallback() had to be modified to walk the parent chain and call the sink again with
+    // en/unitsShort/year and root/unitsShort/year.
+    struct TestPersonUnitsSink : public ResourceSink {
+        typedef std::set<std::string> FoundKeysType;
+        typedef std::map<std::string, FoundKeysType > FoundResourcesType;
```

Comment:
totally optional: this smells like you might have been looking for [std::multimap](https://en.cppreference.com/w/cpp/container/multimap) ...

Reply:
I don't think `std::multimap` does quite what I'm trying to do here.  What I was after was something that would tell me that, for example, `units/duration` (the key in `FoundResourcesType`) had child resources named `one` and `other`, but not `per` or `dnam`.  So I'm mapping from a single string (e.g., `units/duration`) to a collection of strings (`one`, `many`).

Reply:
I think that's what a multimap does -- but, as I wrote earlier, "totally optional".